src/cli/mod.rs

```diff
@@ -0,0 +1,11 @@
+mod config;
+use crate::api_client;
+
+pub use config::{BeaconMethod, CliConfig, Namespace::Beacon};
+
+pub async fn run_cli(client: api_client::Client, args: CliConfig) {
```

Comment:
better to take these by reference, as a default

in general, these types may be too big to want to pass on the stack, so instead we can just pass their address (via the ref-operator `&`)

```suggestion
pub async fn run_cli(client: &Client, args: &CliConfig) {
```